### ISSUE TITLE
timeout = 0, sets timeout to 1800 sec

### ISSUE THREAD

**Issue by jofrep** [logstash-plugins/logstash-filter-aggregate#31]
According to the documentation:
"Timeout:
The default value is 0, which means no timeout so no auto eviction. "

But when I set the timeout to 0 is it set 1800s  

- Logstash Version: 2.3.2
- Aggregate version: 2.1.1
- Operating System: Raspbian GNU/Linux 8 
- Config File:

```
input {
  file {
    path => "/var/log/dnsmasq.log"
    start_position => "beginning"
    type => "dnsmasq"
  }
}

filter {
  if [type] == "dnsmasq" {
    grok {
      match =>  [ "message", "%{SYSLOGTIMESTAMP:reqtimestamp} %{USER:program}\[%{NONNEGINT:pid}\]\: ?(%{NONNEGINT:num} )?%{NOTSPACE:action} %{IP:clientip} %{MAC:clientmac} ?(%{HOSTNAME:clientname})?"]
      match =>  [ "message", "%{SYSLOGTIMESTAMP:reqtimestamp} %{USER:program}\[%{NONNEGINT:pid}\]\: ?(%{NONNEGINT:num} )?%{USER:action}?(\[%{USER:subaction}\])? %{NOTSPACE:domain} %{NOTSPACE:function} %{IP:clientip}"]
      match =>  [ "message", "%{SYSLOGTIMESTAMP:reqtimestamp} %{USER:program}\[%{NONNEGINT:pid}\]\: %{NOTSPACE:action} %{DATA:data}"]
    }

    if [action] =~ "DHCPACK" {
      if ![clientname] {
        mutate {
          add_field => { "clientname" => "No name" }
        }
      }
      aggregate {
        task_id => "%{clientip}"
        code => "map['clientmac'] = event['clientmac']; map['clientname'] = event['clientname'];"
        map_action => "create_or_update"
        # timeout = 0 sets the timeout to the default value 1800 seconds.
        timeout => 172800
      }
    } else if [action] == "query" {
      aggregate {
        task_id => "%{clientip}"
        code => "event['clientmac'] = map['clientmac']; event['clientname'] = map['clientname']"
        map_action => "update"
      }
      if ![clientname] {
        mutate {
          add_field => { "clientname" => "%{clientip}" }
        }
      }
      if ![clientmac] {
        mutate {
          add_field => { "clientmac" => "%{clientip}" }
        }
      }
    } else if [action] == "reply" {
        mutate {
          rename => { "clientip" => "serverip" }
        }
      geoip {
        source => "serverip"
      }
    } else
    {
      drop{}
    }
  }
}
output {
 elasticsearch { hosts => ["localhost:9200"] }
}
```

**Comment by fbaligand** [comment]
Actually, there is an error in documentation.
As you mention, by default, timeout is 1800s.
I will fix documentation to mention that.

It is described elsewhere in documentation, in "How it works" section :
`if no timeout is defined, by default, all maps older than 1800 seconds are automatically deleted` 

This is a memory protection to avoid that aggregate maps accumulate in memory.
If you want an explicit value to say no timeout (ex: -1), I invite you to create a specific issue for this enhancement.

**Comment by fbaligand** [comment]
I just released version 2.1.2.
It clarifies default timeout behaviour.
